app/Http/Controllers/BeatmapsetsController.php

```diff
@@ -255,6 +255,26 @@ public function nominate($id)
         return $beatmapset->defaultDiscussionJson();
     }
 
+    public function love($id)
+    {
+        $beatmapset = Beatmapset::findOrFail($id);
+
+        priv_check('BeatmapsetLove', $beatmapset)->ensureCan();
+
+        $nomination = $beatmapset->love(Auth::user());
+        if (!$nomination['result']) {
+            return error_popup($nomination['message']);
+        }
+
+        BeatmapsetWatch::markRead($beatmapset, Auth::user());
+        (new NotifyBeatmapsetUpdate([
```

Comment:
mark read &notify aren't necessary

Reply:
If you're watching a beatmapset, I'd argue that you'd probably want to be notified if it becomes loved?